### DISCUSSION TITLE
StarUML 사용법

### DISCUSSION THREAD

**Discussion by Lcat12345** [MyeongWoonJang/cpp-study#172] [General]
## 다운로드 링크
https://staruml.io/
## 메인 화면
![image](https://user-images.githubusercontent.com/62102830/212469228-53dd360b-b2ae-486e-b058-965e12293a71.png)
## 도구 상자
<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212469426-c859b9e5-4d3d-43a2-b747-f992f020e9e8.png"></p>

* Class
<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212529696-ab27ac6e-853d-424b-8d04-359013bfe6da.png"></p>

버튼을 클릭하고 빈 곳 아무 데나 클릭을 하면 3구역으로 나눠진 사각형이 만들어진다. 

처음에 이름을 정할 수 있는데 특별히 정하지 않는다면 Class(number)로 만들어진다.

* Interface
<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212529819-8c5383dc-bab8-4596-b18e-b9ac75595b44.png"></p>

원이 하나 만들어지고 이름을 정할수 있는데, 특별히 정하지 않는다면 Interface(number)로 만들어진다.

단, 이렇게 만들면 어떤 동작을 하는지 제대로 알기 힘들다.

그래서 아래와 같은 방법으로 Interface 클래스를 만든다.

1. 원형태로 되어있는 Interface클래스를 우클릭 - Format - StereoType Display를 Icon 에서 label로 바꾼다.

<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212535853-f108514f-e108-44b9-8df7-f60606984196.png"></p>

2. 이름을 더블클릭하여 나오는 버튼을 활용해서 구현 클래스를 만들거나 새로 class를 하나 만들어서 Interface Realization 화살표로 연결한다.

<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212535958-05d87c73-e5fc-44a8-8e02-7ad0601ff91e.png"></p>

완성 예시)
<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212535272-771d4a2f-f657-4b83-879d-7a1fa58cd347.png"></p>

* Asscociation
<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212531040-54bda574-def2-4dd0-b99d-b8fab845728d.png"></p>

연관 관계를 표시하며 실선을 그을 수 있다.

* Directed Association
<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212531168-5a0039ff-2566-4f47-a493-fa1909ead21f.png"></p>

직접 연관관계를 표시하며 실선 화살표를 그을 수 있다.

<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212531568-59000124-fac3-4138-a7ae-2de93bab56af.png"></p>

화살표를 클릭해보면 오른쪽 에디터창에서 속성을 볼수있는데 end1.multiplicity칸에 객체를 몇개 가지는지를 표현할수 있다.

<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212531704-6773bd8a-8be1-40f6-8bc7-c5f021827317.png"></p>

reference를 주의하자

end1.reference class2의 multiplicity칸에 최소 최대값을 넣어보면 

<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212533960-8361bd5c-79a7-4b07-9773-809a0accdc73.png"></p>

* Aggregation
<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212531417-0df45bb6-3732-4c6a-ad76-f81c76f77af5.png"></p>

* Composition
* Dependency
* Generalization
* Interface Realization

## class Format
class는 3개의 구역으로 나뉘어진다.

첫번째는 name, 두번째는 attribute, 세번째는 operation이다.
그러나 설계 단계의 class diagram은 attribute와 operation을 작성하지 않는 경우도 있다.
class를 우클릭 - Format - Suppress (Attributes or Opreations) 

<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212536504-7de40e32-daae-4226-a6c1-5a44ea13c245.png"></p>

3개의 구역으로 나뉘어진 class가 이름만 남게 되었다.

<p align="center"><img src="https://user-images.githubusercontent.com/62102830/212536623-f638e4c5-fe37-4d5d-a95c-e4d760e2bbe2.png"></p>

### 시작하는 방법

빈 공간에 우클릭 -> Add Diagram -> Use Case Diagram

<p align="center"><img src="https://user-images.githubusercontent.com/62102830/215263294-c9ed1c14-9378-45aa-a66c-b529a84c8702.png"></p>

**Comment by Woon-2** [comment]
![캡처](https://user-images.githubusercontent.com/73771162/212688578-f4c6e54b-763e-4c31-a310-28e557f8a5d4.PNG)

제 환경에서도 사용 가능한 것을 확인했습니다!
